### ISSUE TITLE
Never gets to CancelledException block

### ISSUE THREAD

**Issue by DavidBucienski** [JonathanPeterCole/dart-cancellation-token-http#1]
Your package may still work for me even with this unexpected behavior. I working it into my project. I'll let you know how it goes.
The code you requested.
Edit: Changed print statement in http.CancelledException block.
Edit: Corrected comment above try block and added token reset line.

`import 'dart:io';
import 'package:cancellation_token_http/retry.dart';
import 'package:cancellation_token_http/http.dart' as http;

class HttpStreamedResponse {
  final token = http.CancellationToken();

  void stopDownload() {
    token.cancel();
  }

  Future<http.StreamedResponse> httpStreamResponse(Uri url) async {
    if (token.isCancelled) token = ct_http.CancellationToken();
    final client = RetryClient(http.Client());
    final request = http.Request('GET', url);

**// When token is cancelled it throws, Instance of 'CancelledException', from token.cancel()**
**// If token.cancel() is called from within the try block there is an Error in console, best to call it from stopDownload()**
    try {
      http.StreamedResponse response = await client
          .send(request, cancellationToken: token)
          .timeout(const Duration(seconds: 3), onTimeout: () async {
        await InternetAddress.lookup('www.google.com')
            .timeout(const Duration(seconds: 3), onTimeout: () {
          throw 'No Internet Connection';
        });
        throw 'Server Unavailable';
      });

      return response;
**// Will never get to this block**
    } on http.CancelledException {
      print('Will Get Here When Sending New Request');
      throw 'Request Canceled';
    } on SocketException {
      throw 'Check Your Internet Connection';
    } // No finally used here, because it is a StreamedResponse
  }
}`

**Comment by JonathanPeterCole** [comment]
Unfortunately I haven't been able to replicate this yet, although this is only based on unit testing as I haven't had time to test with a real world application yet.

However, I have observed cases where the stack traces after cancellation can be misleading. If a CancelledException isn't caught correctly, the stack trace may show the call stack leading up to the cancellation of the token, rather than the call stack for the request that threw the exception. This can make it appear that the token itself is throwing the exception. 

In an upcoming update to the cancellation_token package I'll be making some improvements to ensure that cancellation stack traces point you to the place in the code where the exception was thrown.

**Comment by radbackup** [comment]
I've been playing with it more, today. Here is what I've seen:
1. Like I said last week the "on http.CancelledException" block is ignored. However, "Instance of 'CancelledException'" is thrown. When token.cancel() is called.
2. But then, when I send a new request the "on http.CancelledException" block is reached immediately. I can no longer download a file.

**Comment by radbackup** [comment]
I gave the [dio package](https://pub.dev/packages/dio) a try and found it has the same behavior as your package. I discovered the reason, I could no longer make a request after cancelling the token is because the token needs to be reset. So I switched back to your package. I like the idea of your package over dio and don't need all the features. Right, now I'm using your package without any issues. Ya, it throws funny, but I catch it on the next layer which is where I need it to go.
